libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -1243,6 +1244,23 @@ void Compass::_probe_external_i2c_compasses(void)
     }
 #endif  // AP_COMPASS_LIS3MDL_ENABLED
 
+#if AP_COMPASS_LIS2MDL_ENABLED
+    // external lis2mdl
+#if AP_COMPASS_LIS2MDL_EXTERNAL_BUS_PROBING_ENABLED
+    FOREACH_I2C_EXTERNAL(i) {
+        ADD_BACKEND(DRIVER_LIS2MDL, AP_Compass_LIS2MDL::probe(GET_I2C_DEVICE(i, HAL_COMPASS_LIS2MDL_I2C_ADDR),
+                    true, ROTATION_YAW_90));
+    }
+#endif
+    // internal lis2mdl
+#if AP_COMPASS_INTERNAL_BUS_PROBING_ENABLED
+    FOREACH_I2C_INTERNAL(i) {
+        ADD_BACKEND(DRIVER_LIS2MDL, AP_Compass_LIS2MDL::probe(GET_I2C_DEVICE(i, HAL_COMPASS_LIS2MDL_I2C_ADDR),
+                    all_external, all_external?ROTATION_YAW_90:ROTATION_NONE));
```

Comment:
why not ROTATION_NONE for both, and fix in hwdef for ones that are different?

Reply:
ok changed